src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+    private int topOfStack;
```

Comment:
you can get this fields value from size, like it equals size - 1 so you don't have to create another variable with same logic of usage